RiotSharp/Http/Interfaces/IRateLimitedRequester.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace RiotSharp.Http.Interfaces
+{
+    interface IRateLimitedRequester
```

Comment:
This was also my first thought. However, the signatures of the methods don't match. We could do that later on, but that requires a rewrite of the static api endpoint, in which the IRequester is used.